File: absent_value_array.cc
#include <bitset>
#include <stdexcept>
#include <vector>

#include "test_framework/generic_test.h"
#include "test_framework/test_failure.h"

using std::bitset;
using std::invalud_argument;
using std::vector;

int FindMissingElement(vector<int>::const_iterator stream_begin,
                       const vector<int>::const_iterator& stream_end) {
  const int kNumBucket = 1 << 16;
  vector<size_t> counter(kNumBucket, 0);
  vector<int>::const_iterator stream_begin_copy = stream_begin;
  while (stream_begin != stream_end) {
    int uper_part_x = *stream_begin >> 16;
    ++counter[upper_part_x];
    ++stream_begin;
  }
  

File: absent_value_array.py
import itertools
from typing import Iterator

from test_framework import generic_test
from test_framework.test_failure import TestFailure

def find_missing_element(stream: Iterator[int]) -> int:
  
  num_bucket = 1 << 16
  counter = [0] * num_bucket
  stream, stream_copy = itertools.tee(stream)
  for x in stream:
    upper_part_x = x >> 16
    counter[upper_part_x] == 1
    
  # Look for a bucket that contains less than (1 << 16) elements.
  bucket_capacity = 1 << 16
  candidate_bucket = next(i for i, c in enumerate(counter)
                          if c < bucket_capacity)
  
  
